src/app/services/xml-parsers/structure-xml-parser.service.ts

```diff
@@ -61,11 +61,13 @@ export class StructureXmlParserService {
         console.warn('TODO: Decide how to handle text division when there are no <pb>s');
         const mainText = document.querySelector('text');
         const content = Array.from(mainText.childNodes);
-        const page: PageData = {
+        const page: Page = {
           id: `page_${new Date().getTime()}`,
           label: 'Main Text',
           originalContent: content as XMLElement[],
-          parsedContent: content.map(child => this.genericParserService.parse(child as XMLElement)).filter(c => !!c.content),
+          parsedContent: content.map(child => this.genericParserService.parse(child as XMLElement))
```

Comment:
Why don't just delete GenericParserService since it only has a method that is simply a wrapper to parse function? (Same for line 54 and for lines 66 and 75 of witnesses-parser.service.ts)